include/execution.hpp

```diff
@@ -838,50 +839,81 @@ namespace std::execution {
+                noexcept(is_nothrow_invocable_v<_Tag, const _Receiver&, _As...>)
+                -> invoke_result_t<_Tag, const _Receiver&, _As...> {
+              return ((_Tag&&) __tag)((const _Receiver&) __self.__op_state_->__rcvr_, (_As&&) __as...);
             }
           };
           _Receiver __rcvr_;
-          connect_result_t<_Sender, __wrap> __state_;
-          __receiver(_Sender&& __sndr, _ReceiverId&& __rcvr)
-            : __rcvr_((_ReceiverId&&) __rcvr)
-            , __state_(connect((_Sender&&) __sndr, __wrap{this}))
+          connect_result_t<_Sender, __receiver> __op_state_;
+          __operation(_Sender&& __sndr, __decays_to<_Receiver> auto&& __rcvr)
+            : __rcvr_((decltype(__rcvr)&&) __rcvr)
+            , __op_state_(connect((_Sender&&) __sndr, __receiver{this}))
           {}
         };
-    }
+    } // namespace __impl
 
-    inline constexpr struct submit_t {
+    inline constexpr struct __submit_t {
       template <receiver _Receiver, sender_to<_Receiver> _Sender>
       void operator()(_Sender&& __sndr, _Receiver&& __rcvr) const noexcept(false) {
-        start((new __impl::__receiver<_Sender, _Receiver>{(_Sender&&) __sndr, (_Receiver&&) __rcvr})->__state_);
+        start((new __impl::__operation<__x<_Sender>, __x<decay_t<_Receiver>>>{
+            (_Sender&&) __sndr, (_Receiver&&) __rcvr})->__op_state_);
       }
-      template <receiver _Receiver, sender_to<_Receiver> _Sender>
-        requires tag_invocable<submit_t, _Sender, _Receiver>
-      void operator()(_Sender&& __sndr, _Receiver&& __rcvr) const
-        noexcept(nothrow_tag_invocable<submit_t, _Sender, _Receiver>) {
-        (void) tag_invoke(submit_t{}, (_Sender&&) __sndr, (_Receiver&&) __rcvr);
+    } __submit {};
+  } // namespace __submit_
+
+  /////////////////////////////////////////////////////////////////////////////
+  // [execution.senders.consumer.start_detached]
+  inline namespace __start_detached {
+    namespace __impl {
+      struct __detached_receiver {
+        friend void tag_invoke(set_value_t, __detached_receiver&&, auto&&...) noexcept {}
+        [[noreturn]]
+        friend void tag_invoke(set_error_t, __detached_receiver&&, auto&&) noexcept {
+          terminate();
+        }
+        friend void tag_invoke(set_done_t, __detached_receiver&&) noexcept {}
+      };
+    } // namespace __impl
+
+    inline constexpr struct start_detached_t {
+      template <sender _Sender>
+        requires tag_invocable<start_detached_t, _Sender>
+      void operator()(_Sender&& __sndr) const
+        noexcept(nothrow_tag_invocable<start_detached_t, _Sender>) {
+        (void) tag_invoke(start_detached_t{}, (_Sender&&) __sndr);
       }
-    } submit {};
-  }
+      template <sender _Sender>
+        requires (!tag_invocable<start_detached_t, _Sender>) &&
+          sender_to<_Sender, __impl::__detached_receiver>
+      void operator()(_Sender&& __sndr) const noexcept(false) {
```

Comment:
I know we are planning to remove support for this in favour of `async_scope` so may be it's not a big issue, but if we were to keep `start_detached()` there should be a way to support custom-allocator for the heap-allocation of the op-state here. e.g. see the `unifex::submit()` implementation.